src/NServiceBus.Core/Serializers/XML/XmlSerializerCache.cs

```diff
@@ -157,24 +157,24 @@ PropertyInfo[] GetAllPropertiesForType(Type t, bool isKeyValuePair)
                 {
                     if (typeof(IList<>).MakeGenericType(args) == prop.PropertyType)
                     {
-                        throw new NotSupportedException("IList<T> is not a supported property type for serialization, use List<T> instead. Type: " + t.FullName + " Property: " + prop.Name);
+                        throw new NotSupportedException($"IList<T> is not a supported property type for serialization, use List<T> instead. Type: {t.FullName} Property: {prop.Name}");
                     }
                     if (typeof(ISet<>).MakeGenericType(args) == prop.PropertyType)
                     {
-                        throw new NotSupportedException("ISet<T> is not a supported property type for serialization, use HashSet<T> instead. Type: " + t.FullName + " Property: " + prop.Name);
+                        throw new NotSupportedException($"ISet<T> is not a supported property type for serialization, use HashSet<T> instead. Type: {t.FullName} Property: {prop.Name}");
                     }
                 }
 
                 if (args.Length == 2)
                 {
                     if (typeof(IDictionary<,>).MakeGenericType(args[0], args[1]) == prop.PropertyType)
                     {
-                        throw new NotSupportedException("IDictionary<T, K> is not a supported property type for serialization, use Dictionary<T,K> instead. Type: " + t.FullName + " Property: " + prop.Name + ". Consider using a concrete Dictionary<T, K> instead, where T and K cannot be of type 'System.Object'");
+                        throw new NotSupportedException($"IDictionary<T, K> is not a supported property type for serialization, use Dictionary<T,K> instead. Type: {t.FullName} Property: {prop.Name}. Consider using a concrete Dictionary<T, K> instead, where T and K cannot be of type \'System.Object\'");
```

Comment:
since I can't see any reason to have this, I'd say let's remove the unnecessary `\` (I saw more usages of it)